src/test/sql/sql_translator_test.cpp

```diff
@@ -367,18 +367,267 @@ TEST_F(SQLTranslatorTest, SelectListAliasUsedInJoin) {
 
   const auto aliases = std::vector<std::string>({"c", "d"});
 
+  // clang-format off
+  const auto expected_lqp_a =
+  ProjectionNode::make(expression_vector(int_float_a, int_float_b),
+    JoinNode::make(JoinMode::Inner, equals_(int_float_b, int_float2_b),
+      AliasNode::make(expression_vector(int_float_a, int_float_b), aliases,
+        stored_table_node_int_float),
+      stored_table_node_int_float2));
+
+  const auto expected_lqp_b =
+  AliasNode::make(expression_vector(int_float_a, int_float_b), aliases,
+    expected_lqp_a);
+  // clang-format on
+
+  EXPECT_LQP_EQ(actual_lqp_a, expected_lqp_a);
+  EXPECT_LQP_EQ(actual_lqp_b, expected_lqp_b);
+}
+
+TEST_F(SQLTranslatorTest, SelectListAliasesDifferentForSimilarColumns) {
+  const auto actual_lqp = compile_query("SELECT a AS a1, a AS a2, a AS a3, b AS b1, b AS b2, b AS b3 FROM int_float");
+
+  const auto aliases = std::vector<std::string>({"a1", "a2", "a3", "b1", "b2", "b3"});
```

Comment:
Not a problem in this case, but generally I'd avoid making test data sorted (unless that is the input you want to test). Otherwise, you might have an algorithm that only works with sorted inputs.